File: ExamTest/app/src/main/java/ru/project/examtest/data/db/MyRoomManager.java
package ru.project.examtest.data.db;

import androidx.room.*;
import ru.project.examtest.data.inerfaces.*;
import ru.project.examtest.data.models.*;

@Database(
        entities = {
                RoomTeacher.class,
                RoomTerms.class,
                RoomTheory.class,
                RoomStudent.class,
                RoomBasicFormula.class,
                RoomDiagrams.class,
                RoomTables.class,
                RoomTasks.class,
                RoomTestOgeP.class
        },
        version = 1
)
public abstract class MyRoomManager extends RoomDatabase {
    public abstract TeacherDAO teacherDAO();
    public abstract TermsDAO termsDAO();
    public abstract TheoryDAO theoryDAO();
    public abstract StudentDAO studentDAO();
    public abstract BasicFormulasDAO basicFormulasDAO();
    public abstract DiagramsDAO diagramsDAO();
    public abstract TablesDAO tablesDAO();
    public abstract TasksDAO tasksDAO();
    public abstract TestOgeP_DAO roomTestOgeP();
}

File: MyAppRoomExample/app/src/main/java/ru/project/roomexample/data/room/MyRoomManager.kt
package ru.project.roomexample.data.room

import androidx.room.Database
import androidx.room.RoomDatabase
import ru.project.roomexample.data.interfaces.UserDAO
import ru.project.roomexample.data.models.UserRoom

@Database(
    entities = [UserRoom::class],
    version = 1
)
abstract class MyRoomManager: RoomDatabase() {
    abstract fun userDAO(): UserDAO
}